Source/Prism/Ioc/IContainerProvider.cs

```diff
@@ -1,11 +1,14 @@
 ﻿using System;
+using System.Collections.Generic;
 
 namespace Prism.Ioc
 {
     public interface IContainerProvider
     {
         object Resolve(Type type);
 
+        object Resolve(Type type, IDictionary<Type, object> parameters);
```

Comment:
I'm not 100% sure about this one as every container implements it differently. Some require the "key" and some don't.  What type of parameters are being passed? What is the "Key" exactly? Are these container specific parameters, or are they application objects?  What is the use case for this?  Also, instead of a dictionary, why not use something like

```
object Resolve(Type type, params (Type Key, object Value)[ ] parameters);
```

Reply:
this is implemented in all of our containers and is available in the rest I've looked at. DryIoc is awesome and just requires the instance... Autofac (which we dropped), Ninject, and Unity all support injecting the instance by type

Reply:
btw, this would eliminate the need for us to add an Android specific target for Prism.DryIoc.Forms and Prism.Unity.Forms and will be very useful for UWP

Reply:
I know each container has some version of this, but those signatures are different. So for example, in Unity, the "Key" is not used which makes for a misleading API. What about my other questions? Can you provide more context around the use of this.